node-package/bin/dukat-cli.js

```diff
@@ -23,18 +28,37 @@ function run(command, args) {
     return child;
 }
 
+function printInvalidJavaHome() {
+    printError("Your \"JAVA_HOME\" environment variable points to an invalid directory.");
+    printError("As of now \"dukat\" requires Java Runtime Environment to be installed.");
+    printError("Either set \"JAVA_HOME\" to a proper Java installation,");
+    printError("or make sure \"java\" is callable from commandline and unset \"JAVA_HOME\".");
+}
+
 function printNoJava() {
-    printError("It looks like you don't have java installed or it's just not reachable from command-line");
+    printError("\"java\" is not callable from commandline and no \"JAVA_HOME\" environment variable is set.");
     printError("As of now \"dukat\" requires Java Runtime Environment to be installed.");
+    printError("Either set \"JAVA_HOME\" to a proper Java installation,");
+    printError("or make sure \"java\" is callable from commandline.");
 }
 
-function guardJavaExists() {
-    exec("java -version", function (error, stdout, stderr) {
-        if (error) {
+function findJavaExecutable() {
+    var javaHome = process.env.JAVA_HOME
+    var result = javaHome ? path.join(javaHome, "bin", "java") : "java"
+    try {
+        execSync('"' + result + "\" -version", {
+            stdio: ['pipe', 'ignore', 'pipe']
+        });
+    } catch (error) {
+        printError(error)
+        if (javaHome) {
+            printInvalidJavaHome();
+        } else {
             printNoJava();
-            process.exit(error.code);
         }
-    });
+        process.exit(1);
```

Comment:
It seems to me no harm to pass actual error.code rather than 1 - is there any reasoning behind this decision?

Reply:
The reason was, that `error.code` is `undefined`, at least in the cases I looked at it.
There is a property `error.status`, but I was unsure whether this represents the exit code.
Thus I decided to just always use 1.